internal/app/machined/pkg/runtime/v1alpha1/platform/packet/packet.go

```diff
@@ -77,6 +79,24 @@ func (p *Packet) Name() string {
 // Configuration implements the platform.Platform interface.
 // nolint: gocyclo
 func (p *Packet) Configuration(ctx context.Context) ([]byte, error) {
+	ntpCtx, ntpCtxCancel := context.WithCancel(ctx)
+	defer ntpCtxCancel()
+
+	go func() {
+		ntpClient, err := ntp.NewNTPClient()
+		if err != nil {
+			log.Printf("failed to build one-time ntp client")
+
+			return
+		}
+
+		if err = ntpClient.QueryAndSetTime(ntpCtx); err != nil {
+			if !stderrors.Is(err, context.Canceled) {
+				log.Printf("failed to do one-time time sync: %s", err)
+			}
+		}
+	}()
+
```

Comment:
Feels like maybe we can put this in a more generic place? Maybe in the task that calls the platform interface `Configuration` method?

Reply:
yep, I think it makes perfect sense to do that there.